lib/api/v3/placeholder_users/placeholder_users_api.rb

```diff
@@ -0,0 +1,59 @@
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+module API
+  module V3
+    module PlaceholderUsers
+      class PlaceholderUsersAPI < ::API::OpenProjectAPI
+        resources :placeholder_users do
+          after_validation do
+            authorize_any %i[manage_placeholder_user], global: true
```

Comment:
This will lead to only users having the `manage_placeholder_user` permission to be able to query the show endpoint for placeholder users which is not as intended. 

In this case, you should be able to simply remove that check as the contracts will catch an unauthorized access to patch/delete/post.

Reply:
If I remove the check, all users will be able to see all placeholder users. Do we need to update the query to have a base scope of `PlaceholderUser.visible` instead?

Reply:
Good thinking. Initially I thought that it would not be necessary but it might make sense not rather disclose less information. So how about we implement the same as you did in https://github.com/opf/openproject/pull/9035 for placeholder users. On the other hand, we don't do the same for groups and also not on the UI (where we should be doing the same for users at the very least)
Let's focus on the placeholders for now where we add the scope and open up a ticket for the other parts.

Reply:
I'm merging the PR now and we can add the visible scope separately.